airflow/contrib/operators/mysql_to_gcs.py

```diff
@@ -120,15 +123,20 @@ def _write_local_data_files(self, cursor):
             names in GCS, and values are file handles to local files that
             contain the data for the GCS objects.
         """
-        schema = list(map(lambda schema_tuple: schema_tuple[0], cursor.description))
+        field_names = list(map(lambda schema_tuple: schema_tuple[0], cursor.description))
+        mysql_types = list(map(lambda schema_tuple: schema_tuple[1], cursor.description))
+        byte_fields = [self.is_binary(t, f) for t, f in zip(mysql_types, cursor.description_flags)]
+
         file_no = 0
-        tmp_file_handle = NamedTemporaryFile(delete=True)
+        tmp_file_handle = NamedTemporaryFile(mode='w', delete=True)
         tmp_file_handles = {self.filename.format(file_no): tmp_file_handle}
 
         for row in cursor:
-            # Convert datetime objects to utc seconds, and decimals to floats
-            row = map(self.convert_types, row)
-            row_dict = dict(zip(schema, row))
+            # Convert datetime objects to utc seconds, decimals to floats, and binaries
+            # to base64-encoded strings
+            row_dict = {}
+            for name, (value, is_binary) in zip(field_names, zip(row, byte_fields)):
```

Comment:
This might be clearer:

```
>>> a=[0,1]
>>> b=[2,3]
>>> c=[4,5]
>>> zip(a, b, c)
[(0, 2, 4), (1, 3, 5)]
>>> for foo in zip(a, b, c):
...     print(foo)
... 
(0, 2, 4)
(1, 3, 5)
```